lib/phantom_script.js

```diff
@@ -2,48 +2,84 @@
-        || Package["jazeee:spiderable-longer-timeout"] === undefined
-        || Package["jazeee:spiderable-longer-timeout"].Spiderable === undefined
-        || !Package["jazeee:spiderable-longer-timeout"].Spiderable._initialSubscriptionsStarted ) {
-      return false;
-    }
-    if( !(Meteor.isRouteComplete || Meteor.isReadyForSpiderable ) ) {
-        // We only need one of these flags set in order to proceed. I may deprecate Meteor.isRouteComplete after 2015-12-31
+var renderPage = function(url) {
+  var intervalId = false;
+  var isReady = function () {
+    return page.evaluate(function () {
+      if (typeof Meteor === 'undefined'
+          || Meteor.status === undefined
+          || !Meteor.status().connected) {
         return false;
-    }
-    if (typeof Tracker === 'undefined' || typeof DDP === 'undefined'){
+      }
+      if (typeof Package === 'undefined'
+          || Package["jazeee:spiderable-longer-timeout"] === undefined
+          || Package["jazeee:spiderable-longer-timeout"].Spiderable === undefined
+          || !Package["jazeee:spiderable-longer-timeout"].Spiderable._initialSubscriptionsStarted ) {
         return false;
+      }
+      if( !(Meteor.isRouteComplete || Meteor.isReadyForSpiderable ) ) {
+          // We only need one of these flags set in order to proceed. I may deprecate Meteor.isRouteComplete after 2015-12-31
+          return false;
+      }
+      if (typeof Tracker === 'undefined' || typeof DDP === 'undefined'){
+          return false;
+      }
+      Tracker.flush();
+      if(!DDP._allSubscriptionsReady()){
+        return false;
+      }else if(Spiderable.redirect){
+        return {redirectTo: Spiderable.redirect};
+      }else{
+        return true;
+      }
+    });
+  };
+
+  var dumpPageContent = function () {
+    var output = page.content;
+    output     = output.replace(/<script[^>]+>(.|\n|\r)*?<\/script\s*>/ig, '');
+    output     = output.replace('<meta name="fragment" content="!">', '');
+    console.log(JSON.stringify({
+      status: status, 
+      headers: headers,
+      html: output
+    }));
+  };
+
+  page.onResourceReceived = function(response) {
+    if(response.url === url || response.redirectURL === url){
+      status  = response.status;
+      headers = response.headers;
+    }
+    if(response.status != 200){
+      status  = response.status;
+    }
+  };
+
+  page.open(url, function(status) {
+    if (status === 'fail'){
+      phantom.exit();
     }
-    Tracker.flush();
-    return DDP._allSubscriptionsReady();
   });
-};
 
-var dumpPageContent = function () {
-  var out = page.content;
-  out = out.replace(/<script[^>]+>(.|\n|\r)*?<\/script\s*>/ig, '');
-  out = out.replace('<meta name="fragment" content="!">', '');
-  console.log(out);
+  intervalId = setInterval(function() {
+    var renderStatus = isReady();
+    if (renderStatus === true || status != 200) {
+      clearInterval(intervalId);
+      setTimeout(function(){
```

Comment:
This doesn't seem correct.
We should not need to do this and it will be fragile under server load.

If you are finding that you have to wait 50 msec before dumping page content, then the isReady is not ready, and you are masking the issue by waiting for 50 msec.